### ISSUE TITLE
[Feature request] "Surround" support in RMarkdown files

### ISSUE THREAD

**Issue by pat-s** [Ikuyadeu/vscode-R#260]
Even though I mostly use vim-surround, I noticed that there is no native "surround" support for R within vscode by default (i.e. not in vim mode).

Seems like this needs to be added on a [per language base](https://stackoverflow.com/questions/42363030/visual-studio-code-surround-with).

**Comment by andycraig** [comment]
Hi @pat-s, thank you for the suggestion. Let's say I have a file `temp.R` containing just the word `dat`. If I select the whole word and hit `"` I get:

```
"dat"
```

Similarly for `(`, `{` and `[`.

If I select the word `dat` and hit `'`, I get:

```
'
```

So, it seems we don't support single quote surrounding at the moment, but do support double-quotes, parenthesis, braces and brackets.

Is that what you're referring to?

**Comment by pat-s** [comment]
> So, it seems we don't support single quote surrounding at the moment, but do support double-quotes, parenthesis, braces and brackets.

Single quotes also work for me. Let me recall where I experienced this - it might be that I missed this in an Rmd doc.

There I could reproduce the issue right now again. In `.md` files I have surround support for `[` but this is also missing in `.Rmd`. So there is no surround support in `.Rmd` at all currently?

I have updated the title to point to Rmd files only - `.R` scripts seem to function correctly.
